client/src/components/Upload/utils.js

```diff
@@ -1,7 +1,149 @@
 /*
  * Utilities for working with upload data structures.
  */
+import axios from "axios";
+import { getAppRoot } from "onload/loadConfig";
+import { rethrowSimple } from "utils/simple-error";
 
+const AUTO_EXTENSION = {
+    id: "auto",
+    text: "Auto-detect",
+    description:
+        "This system will try to detect the file type automatically. If your file is not detected properly as one of the known formats, it most likely means that it has some format problems (e.g., different number of columns on different rows). You can still coerce the system to set your data to the format you think it should be.  You can also upload compressed files, which will automatically be decompressed.",
+};
+const DEFAULT_DBKEY = "?";
+const DEFAULT_EXTENSION = "auto";
+
+/**
+ * Local cache.
+ */
+let _cachedDatatypes;
+let _cachedDbKeys;
+
+/*
+ * Local helper utilities.
+ */
+function dbKeySort(defaultDbKey) {
+    return (a, b) => {
+        if (a.id == defaultDbKey) {
+            return -1;
+        }
+        if (b.id == defaultDbKey) {
+            return 1;
+        }
+        return a.text > b.text ? 1 : a.text < b.text ? -1 : 0;
+    };
+}
+
+async function loadDbKeys() {
+    if (_cachedDbKeys) {
+        return Promise.resolve(_cachedDbKeys);
+    }
+    const url = `${getAppRoot()}api/genomes`;
```

Comment:
Can you please use the typescript fetcher for the endpoints that are in FastAPI, so everything moved to here.